### PR TITLE
contents: algorithm cheatsheet typos and suggestions

### PR DESCRIPTION
**Pull request by jomosz**
Suggestions:
- Heap page: Potentially mention that for python heapq you can push the inverse value to find the max
- Tree page: Traversing balanced tree is O(h) space

Please correct me if I am wrong!

### CODE REVIEW

File: contents/algorithms/study-cheatsheet.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ For each topic, you can expect to find:
 1. Time complexities cheatsheet
 1. Things to look out for during interviews
 1. Corner cases
-1. Useful techniques vRecommended questions to practice
+1. Useful techniques with recommended questions to practice
```
Comment (jomosz) [inline contents/algorithms/study-cheatsheet.md:29]: Not quite sure what it should be

Comment (yangshun) [reply contents/algorithms/study-cheatsheet.md:29]: This is correct :)

**Review by yangshun — APPROVED** [review]
@jomosz I'd like to extend my sincerest thanks to you for fixing the typos. Thank you so so much.

**Comment by yangshun** [comment]
@jomosz I called out this great PR here - https://twitter.com/techinterviewhb/status/1525261664773279744

**Comment by jomosz** [comment]
@yangshun Happy to help! This has been a valuable resource! I love the new Grind 75 interface and how it has streamlined my interview prep 😄

**Pull request merged**